lcUI/cadmdichild.cpp

```diff
@@ -385,4 +385,15 @@ void CadMdiChild::cancelCurrentOperations() {
 
 lc::UndoManager_SPtr CadMdiChild::undoManager() const {
     return _undoManager;
+}
+
+std::shared_ptr<LCViewer::Cursor> CadMdiChild::cursor() const {
+	return _cursor;
+}
+
+unsigned int CadMdiChild::id() {
```

Comment:
If _id is only 16 bits that makes 65536 new documents, that's possible if someone never close the program.
I think the better solution is to reuse the id of closed documents, then all the windows would have to be opened at the same time to obtain a collision. It should be possible by calling a Lua function with Qt destroyed event